### PR TITLE
Second attempt at updating all modules

### PR DESCRIPTION
**Pull request by maxulysse**
Since I'm going crazy with https://github.com/nf-core/sarek/pull/440
This is another attempt at it starting from scratch with tiny steps to figure out when and where and what and maybe why it is failing.

### CODE REVIEW

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -120,7 +120,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install nf-core
+          pip install nf-core git+https://github.com/nf-core/tools.git@dev
```
Comment (maxulysse) [inline .github/workflows/linting.yml:123]: Just adding that to have the linting from dev

File: conf/test.config [hunk 2]
```diff
@@ -42,7 +42,7 @@ profiles {
         params.step                = 'annotate'
     }
     pair {
-        params.input               = 'https://raw.githubusercontent.com/nf-core/test-datasets/sarek/testdata/csv/tiny-manta-https.csv'
+        params.input               = "${baseDir}/tests/csv/3.0/fastq_pair.csv"
     }
     prepare_recalibration {
         params.input               = 'https://raw.githubusercontent.com/nf-core/test-datasets/sarek/testdata/csv/tiny-mapped-normal-https.csv'
```
Comment (maxulysse) [inline conf/test.config:48]: Need to update for the tests

File: tests/test_save_bam_mapped.yml [hunk 3]
```diff
@@ -5,26 +5,23 @@
...
   files:
     - path: results/multiqc
-    - path: results/preprocessing/1234N/mapped/1234N.bam
-    - path: results/preprocessing/1234N/mapped/1234N.bam.bai
-    - path: results/preprocessing/1234N/markduplicates/1234N.md.cram
-    - path: results/preprocessing/1234N/markduplicates/1234N.md.cram.crai
-    - path: results/preprocessing/1234N/recal_table/1234N.recal.table
-    - path: results/preprocessing/1234N/recalibrated/1234N.recal.cram
-    - path: results/preprocessing/1234N/recalibrated/1234N.recal.cram.crai
+    # - path: results/preprocessing/test/mapped/test-test_L1.bam
```
Comment (maxulysse) [inline tests/test_save_bam_mapped.yml:8]: Need to keep the bam file for samples with a single pair of fastq

Comment (FriederikeHanssen) [reply tests/test_save_bam_mapped.yml:8]: Do we need to create the single end cram file? I can do that if you need it

Comment (maxulysse) [reply tests/test_save_bam_mapped.yml]: No, no issue there

**Review by FriederikeHanssen — APPROVED** [review]
Amazing ❤️

File: .github/workflows/linting.yml (hunk 1 above)
Comment (maxulysse) [reply .github/workflows/linting.yml:123]: not sure that @drpatelh would approve

**Pull request merged**